src/run/storage/request-scoped-in-memory-cache.cts

```diff
@@ -25,44 +25,71 @@ export function setInMemoryCacheMaxSizeFromNextConfig(size: unknown) {
   }
 }
 
-const estimateBlobSize = (valueToStore: BlobType | null | Promise<unknown>): number => {
+type PositiveNumber = number & { __positive: true }
+const isPositiveNumber = (value: unknown): value is PositiveNumber => {
+  return typeof value === 'number' && value > 0
+}
+
+const BASE_BLOB_SIZE = 25 as PositiveNumber
+
+const estimateBlobKnownTypeSize = (
+  valueToStore: BlobType | null | Promise<unknown>,
+): number | undefined => {
   // very approximate size calculation to avoid expensive exact size calculation
   // inspired by https://github.com/vercel/next.js/blob/ed10f7ed0246fcc763194197eb9beebcbd063162/packages/next/src/server/lib/incremental-cache/file-system-cache.ts#L60-L79
   if (valueToStore === null || isPromise(valueToStore) || isTagManifest(valueToStore)) {
-    return 25
+    return BASE_BLOB_SIZE
   }
   if (isHtmlBlob(valueToStore)) {
-    return valueToStore.html.length
+    return BASE_BLOB_SIZE + valueToStore.html.length
+  }
+
+  if (valueToStore.value?.kind === 'FETCH') {
+    return BASE_BLOB_SIZE + valueToStore.value.data.body.length
+  }
+  if (valueToStore.value?.kind === 'APP_PAGE') {
+    return (
+      BASE_BLOB_SIZE + valueToStore.value.html.length + (valueToStore.value.rscData?.length ?? 0)
+    )
+  }
+  if (valueToStore.value?.kind === 'PAGE' || valueToStore.value?.kind === 'PAGES') {
+    return (
+      BASE_BLOB_SIZE +
+      valueToStore.value.html.length +
+      JSON.stringify(valueToStore.value.pageData).length
+    )
   }
-  let knownKindFailed = false
+  if (valueToStore.value?.kind === 'ROUTE' || valueToStore.value?.kind === 'APP_ROUTE') {
+    return BASE_BLOB_SIZE + valueToStore.value.body.length
+  }
+}
+
+const estimateBlobSize = (valueToStore: BlobType | null | Promise<unknown>): PositiveNumber => {
+  let knownTypeFailed = false
```

Comment:
Technically, you could omit this var now that you're using `estimateBlobKnownTypeSizeError`. It's only used in the warning message below, which could be replaced by checking if `estimateBlobKnownTypeSizeError` is not `undefined`, but it's not a blocker, just something to consider next time you're touching the code here.

Reply:
I'll adjust now - it's unlikely this will be done as follow up

Reply:
https://github.com/opennextjs/opennextjs-netlify/pull/2819/commits/f1ab6281390205b32b23960a79306dc33cb2d5ad dropped that extra unneded variable. Also added calculated size to warning just for it to have a bit more information